fix(chartutil): Scaffold chart fails to deploy pod

hickeyma opened the pull request:
**What this PR does / why we need it**:
When you try to install a scaffold chart OOTB after a `helm create`, a subsequent install fails on the pod deployment with `InvalidImageName` as follows:

```console
default       pod/mychart-1556529386-576b45f6fb-w8vm4     0/1     InvalidImageName   0          31h
``` 

This is because the image name is missing the version.  The following line is used to set the image in `deployment.yaml`: `image: "{{ .Values.image.repository }}:{{ .Release.AppVersion }}"`. It seems that the `Release` object `AppVersion` property does not exist. Therefore, it gets rendered as follows:
```console
[....]
spec:
      containers:
        - name: mychart
          image: "nginx:"
[...]
```
This PR is to fix this issue.

**Special notes for your reviewer**:
Using `AppVersion` from the chart object and also setting the version to `stable` as the stable version of nginx is used. Not sure if this different to the approach that was envisaged for #5553. 

**If applicable**:
- [ ] this PR contains documentation
- [ ] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

hickeyma committed:
Fix the image field for the scaffold chart application

Signed-off-by: Martin Hickey <martin.hickey@ie.ibm.com>

bacongobbler commented on pkg/chartutil/create.go:
```diff
@@ -72,8 +72,9 @@ type: application
-appVersion: 0.1.0
+# incremented each time you make changes to the application. Depending on the application,
+# 'stable' can signify the latest stable version of the application.
+appVersion: stable
```
Is there a reason this was changed?

hickeyma replied:
The app being deployed is nginx and we want the stable version.

mattfarina replied:
This get's into something we should talk about more. "stable" is an [nginx tag](https://hub.docker.com/_/nginx/?tab=tags) for the latest stable release. Since the chart uses the `appVersion` as the tag this will work for nginx.

But, this is scaffolding out a chart and guiding end users. This isn't about doing nginx well. Where do we want to guide people and do we want to guide them in a direction like this?

On the charts repo we have a policy against using moving tags. This is not an allowed behavior because expectations can change out from under users. For example, a user could install a chart into QA with one version. It checks out. During the QA period a new stable release comes out. When the user goes to deploy to production a different version is installed. This causes an issue with reproducibility.

In the case of nginx the changes have also been minor versions that should not break compatibility. But, this doesn't always need to be the case. What if nginx 2.0.0 comes out? For these occasions the above problem is worse.

This makes me not want to use moving tags for releases.

hickeyma replied:
@mattfarina Good point, thanks.
I was trying to mirror previous capability with using the 'stable' tag.

I would propose then to set it as follows:
`appVersion: 1.16.0` to define the version of the nginx app that is being deployed.

How does this sound?

bacongobbler replied:
SGTM

hickeyma replied:
Thanks @bacongobbler @mattfarina for the feedback. Update and ready for review again.

hickeyma committed:
Change the nginx app version to a set tag

Want to avoid moving tags like using 'stable'. Therefore,
specify the specifc nginx version/tag.

Update from comment review:
- https://github.com/helm/helm/pull/5662#discussion_r280122531

Signed-off-by: Martin Hickey <martin.hickey@ie.ibm.com>
